driver/scsi_operation.c

```diff
@@ -548,25 +548,60 @@ WnbdModeSense(_In_ PWNBD_DISK_DEVICE Device,
 }
 
 UCHAR
-WnbdReportLuns(_In_ PSCSI_REQUEST_BLOCK Srb) {
+WnbdReportLuns(_In_ PWNBD_EXTENSION DeviceExtension,
+    _In_ PSCSI_REQUEST_BLOCK Srb)
+{
+    WNBD_LOG_DEBUG(": Enter");
+    ASSERT(DeviceExtension);
     ASSERT(Srb);
 
+    UCHAR SrbStatus = SRB_STATUS_SUCCESS;
     PVOID DataBuffer = SrbGetDataBuffer(Srb);
     ULONG DataTransferLength = SrbGetDataTransferLength(Srb);
 
     if (NULL == DataBuffer) {
-        return SRB_STATUS_INTERNAL_ERROR;
+        SrbStatus = SRB_STATUS_INTERNAL_ERROR;
+        goto Exit;
```

Comment:
We don't perform any cleanup in ``Exit``, why not just return directly?

Reply:
I've added the Exit label just for logging the exit code of the function, but I guess that could be done after the function call within the caller.